grpc/stream_test.go

```diff
@@ -0,0 +1,264 @@
+package grpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/grafana/xk6-grpc/grpc/testutils/grpcservice"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/metrics"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"gopkg.in/guregu/null.v3"
+)
+
+type FeatureExplorerStub struct {
+	grpcservice.UnimplementedFeatureExplorerServer
+
+	getFeature   func(ctx context.Context, point *grpcservice.Point) (*grpcservice.Feature, error)
+	listFeatures func(rect *grpcservice.Rectangle, stream grpcservice.FeatureExplorer_ListFeaturesServer) error
+}
+
+func (s *FeatureExplorerStub) GetFeature(ctx context.Context, point *grpcservice.Point) (*grpcservice.Feature, error) {
+	if s.getFeature != nil {
+		return s.getFeature(ctx, point)
+	}
+
+	return nil, status.Errorf(codes.Unimplemented, "method GetFeature not implemented")
+}
+
+func (s *FeatureExplorerStub) ListFeatures(rect *grpcservice.Rectangle, stream grpcservice.FeatureExplorer_ListFeaturesServer) error {
+	if s.listFeatures != nil {
+		return s.listFeatures(rect, stream)
+	}
+
+	return status.Errorf(codes.Unimplemented, "method ListFeatures not implemented")
+}
+
+func TestStream(t *testing.T) {
```

Comment:
What is the vision for this test?
Do we plan to collect here all the tests for the stream method? I think we should avoid having this pattern again, in the long term it is complex to maintain (the current `ws` and `grpc` are already an example of it, if we modify something then we get a vast set of failing tests). I would prefer if we split this across more specific and more scoped tests.

Reply:
All sub-tests here are about the `Stream` class and its functionality.

But for sure, I'll agree that the maintenance (and even reading) of this is bad. So let me break it into pieces.